app/vmalert/web.go

```diff
@@ -32,9 +32,9 @@ func initLinks() {
 		{fmt.Sprintf("api/v1/alert?%s=<int>&%s=<int>", paramGroupID, paramAlertID), "get alert status by group and alert ID"},
 
 		// system links
-		{"/flags", "command-line flags"},
-		{"/metrics", "list of application metrics"},
-		{"/-/reload", "reload configuration"},
+		{"flags", "command-line flags"},
+		{"metrics", "list of application metrics"},
+		{"-/reload", "reload configuration"},
```

Comment:
This won't work for /flags and /metrics if vmalert is accessed through vmselect. 
These two handlers are served by `lib/httpserver`. I think we can try to update the lib and make these handlers as public methods. And then add routing to vmalert and call these public methods if `/vmalert/flags` or `/vmalert/metrics` is requested.

Reply:
There is no need in proxying requests to `/flags` and `/metrics` from `vmselect` to `vmalert`. Only requests to `/rules`, `/api/v1/rules`, `/alerts`, `/api/v1/alerts` and `/vmalert/*` paths must be proxied.

Reply:
yes, but the links are still present and clickable when visiting through the vmselect. However, these links are working neither for vmselect nor for reverse-proxy.

Reply:
Then it is better to just hide the system urls when `vmalert` is requested via a proxy such as `vmselect` with `-vmalert.proxyURL` command-line flag specified.